platform/disk/sfdisk_partitioner.go

```diff
@@ -56,7 +56,7 @@ func (p sfdiskPartitioner) Partition(devicePath string, partitions []Partition)
 	}
 
 	partitionRetryable := boshretry.NewRetryable(func() (bool, error) {
-		_, _, _, err := p.cmdRunner.RunCommandWithInput(sfdiskInput, "sfdisk", "-uM", devicePath)
+		_, _, _, err := p.cmdRunner.RunCommandWithInput(sfdiskInput, "sfdisk", "-uS", devicePath)
```

Comment:
I don't think this is a correct change. This would change the units from Megabytes to Sectors. And since this command appears to be performing partitioning, changing the units without changing the numeric value (which seems to be converted to meegabytes on line 49) would result in a partition of wrong/unintended size.

Also, the `-uS` option also appears to be deprecated, at least on some platforms:
https://man.cx/?page=sfdisk(8)

So if we have to change from megabytes to something else, it would probably be better (if possible) to convert to the default unit (cylinders), instead of a deprecated unit (sectors).

Reply:
Also, this is the Bosh Agent code, so whatever change needs to be compatible with all stemcells, not just RHEL stemcells.